staging/src/k8s.io/apimachinery/pkg/fields/selector_test.go

```diff
@@ -395,3 +395,17 @@ func TestTransform(t *testing.T) {
 	}
 
 }
+
+func TestNilSelectorToString(t *testing.T) {
+	var nilHasTerm *hasTerm = nil
+	var nilNotHasTerm *notHasTerm = nil
+	testCases := []Selector{
+		nilHasTerm,
+		nilNotHasTerm,
+	}
+	for i, tc := range testCases {
+		if tc.String() != "" {
+			t.Errorf("[%d] Nil select String should return empty string, got: %s", i, tc.String())
```

Comment:
The question is whether a "" is valid every place a term can be, not whether they emit "" after your change.

Reply:
reverted it too